charmcraft.yaml

```diff
@@ -28,13 +28,15 @@ bases:
   - build-on:
     - name: ubuntu
       channel: "22.04"
-      architectures: ["amd64"]
+      architectures: ["amd64", "arm64"]
```

Comment:
I am not sure how this work, it seems to me the charm can be built on amd64 **or** arm64, and run on amd64 and arm64. This is not documented in the doc https://discourse.charmhub.io/t/charmcraft-bases-provider-support/4713?_gl=1*16ybp6h*_ga*NjQ2OTc2MTE4LjE2OTE5OTQ2MDY.*_ga_5LTL1CNEJM*MTcyMzUxOTYxMy42OS4wLjE3MjM1MTk2MTMuNjAuMC4w. Can I confirm the behavior?

Reply:
> it seems to me the charm can be built on amd64 or arm64, and run on amd64 and arm64

Correct.

It is documented on that doc, although it's not very clear that it applies to build-on as well as run-on values:

> architectures: [list of architecture strings], defaults to [machine-architecture] (e.g. [“amd64”,“riscv64”])

Reply:
are we expect to build a charm on amd64, and run on arm64?

Reply:
Yes - there is no build process in the charm, so the build architecture doesn't matter.

Reply:
legacy charm 👍🏼 in that case, wouldn't it make sense to build on amd64 and run-on any arches? (i.e. amd64 and arm64)

Reply:
It doesn't matter.  In this case, we need to build it on arm64 too in the functional test workflow running on arm64.  Alternatively we _could_ build it on amd64, save it as an artifact, then download it on an arm64 machine for the arm64 func tests.  But it's easier this way.

We don't need to restrict building the charm on amd64 only.

Reply:
Yeah, thanks, that's fine I guess since the in workflow, we build on a different arch, we need to have both amd64 and arm64 anyways (otherwise it won't build when you run on arm64 machine). Just want to check if my understanding is correct.

Reply:
Yep that's right. :)

Reply:
> and run-on any arches

Ah yes, we could do that.  But then we're advertising that it runs on all the architectures, and then we really should be testing it on all architectures too.